# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/manasa0906/manasa-flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane:CTR/Engagement Oppurtunity Scoring
I selected the CTR/Engagement Oppurtunity Scoring lane because it focuses on identifying pages with the greatest oppurtunity to improve your engagement.The goal is to use data to prioritize which pages should be optimized first instead of relying on manual guesses.This project can help SEO or content teams make better decisions by ranking pages based on their potential improvement.

In [26]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: An SEO analyst or content manager decides which pages should be prioritized for optimization.

Action: Based on the model’s recommendations, the team can improve page titles, descriptions, keywords, and content to increase click-through rate (CTR) and user engagement.

Cost of a wrong recommendation: If the model recommends the wrong pages, valuable optimization time may be wasted, and pages with real improvement potential could be ignored, resulting in lost traffic and engagement.

Why ML can help: The dataset contains many pages and multiple features, making it difficult to identify the best opportunities manually. Machine learning can recognize patterns in historical data and rank pages according to their likelihood of benefiting from optimization.

In [27]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [28]:
!git clone https://github.com/manasa0906/manasa-flyrank-internship.git

fatal: destination path 'manasa-flyrank-internship' already exists and is not an empty directory.


In [29]:
!find manasa-flyrank-internship -name "*.csv"

manasa-flyrank-internship/data/raw/content_refresh_anonymized.csv
manasa-flyrank-internship/outputs/refresh_queue_sample.csv


In [30]:
import pandas as pd

df = pd.read_csv("manasa-flyrank-internship/data/raw/content_refresh_anonymized.csv")

df.head()
print("Rows:", len(df))
print("Columns:", len(df.columns))
print(df.describe(include="all"))

Rows: 30000
Columns: 44
                  content_id          client_id  search_volume   competition  \
count                  30000              30000   27532.000000  27532.000000   
unique                 30000                 32            NaN           NaN   
top     content_6880eb215048  client_19581e27de            NaN           NaN   
freq                       1               7008            NaN           NaN   
mean                     NaN                NaN     158.882391      0.146954   
std                      NaN                NaN    1518.270825      0.285241   
min                      NaN                NaN       0.000000      0.000000   
25%                      NaN                NaN       0.000000      0.000000   
50%                      NaN                NaN      10.000000      0.000000   
75%                      NaN                NaN      20.000000      0.130000   
max                      NaN                NaN   74000.000000      1.000000   

       competit

I explored the starter dataset to understand whether it is suitable for the CTR / Engagement Opportunity Scoring lane. The dataset contains 30,000 rows and 44 columns, providing enough information to analyze page performance. The search_volume field has 27,532 non-null values, indicating that most records contain useful search demand information, although some values are missing. The maximum search volume is 74,000, showing a wide range of keyword popularity. These observations suggest that the dataset is large enough to support meaningful analysis and help prioritize pages for optimization.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project is designed to support decision-making by identifying pages that may have opportunities to improve engagement. The results can highlight patterns and help prioritize pages for review, but they do not prove that changing a page will definitely increase traffic or CTR. The recommendations should be used as guidance and validated by SEO or content experts before taking action.

In [31]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.